src/dtsh/rich/session.py

```diff
@@ -79,9 +79,11 @@ def create_batch(
                 batch_is = DTShInputFile(batch)
             except DTShInputFile.Error as e:
                 raise DTShError(str(e)) from e
+        elif isinstance(batch, List):
+            # Batch commands from cli arguments.
+            batch_is = DTShInputCmds(batch)
         else:
-            # Batch commands.
-            raise NotImplementedError("create_batch(CMD)")
+            raise NotImplementedError("create_batch({})".format(repr(type(batch))))
```

Comment:
Could we:

```suggestion
            raise NotImplementedError(f"create_batch({repr(type(batch))})")
```

To be honest, I'm personally fine with what you wrote, but the linter compains (C0209, consider-using-f-string) and I don't want to entirely disable C0209: sometimes this also shows for legit reasons.